packages/ui/components/Badge/index.js

```diff
@@ -15,76 +15,92 @@ const {
 const isWeb = Platform.OS === 'web'
 
 const ICON_SIZES = {
-  m: 'xs',
+  s: 'xs',
+  m: 's',
   l: 'm'
 }
 
 function Badge ({
+  style,
+  badgeStyle,
   children,
   color,
   label,
   icon,
   position,
-  size
+  size,
+  variant,
+  max
 }) {
   if (!colors[color]) console.error('Badge component: Color for color property is incorrect. Use colors from $UI.colors')
 
   const [right, setRight] = useState(0)
 
+  let _label = label
+
+  if (label > max) {
+    _label = max + '+'
+  }
+
   function onLayout (event) {
     const { width } = event.nativeEvent.layout
     width && setRight(Math.ceil(width / 2) * -1)
   }
```

Comment:
попробуй onLayout повесить на .root и удаляй то что тут для веба сделано

```js
function onLayout (event) {
  const { width } = event.nativeEvent.layout
  setRight(Math.ceil(width / 2) * -1)
 }

// waiting for `right` to prevent badge blinking
return pug`
  Div.root(onLayout=onLaoyut)
    if right
      ...
`
```